src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt

```diff
@@ -115,16 +115,31 @@ implication        ,X86   ,AVXVNNI              ,AVX2
 implication        ,X86   ,MOVBE                ,SSE42
 implication        ,X86   ,X86Serialize         ,X86Base
 implication        ,X86   ,AVX512F              ,AVX2
+implication        ,X86   ,AVX512F              ,FMA
 implication        ,X86   ,AVX512F_VL           ,AVX512F
 implication        ,X86   ,AVX512CD             ,AVX512F
+implication        ,X86   ,AVX512CD_VL          ,AVX512CD
 implication        ,X86   ,AVX512CD_VL          ,AVX512F_VL
 implication        ,X86   ,AVX512BW             ,AVX512F
+implication        ,X86   ,AVX512BW_VL          ,AVX512BW
 implication        ,X86   ,AVX512BW_VL          ,AVX512F_VL
 implication        ,X86   ,AVX512DQ             ,AVX512F
+implication        ,X86   ,AVX512DQ_VL          ,AVX512DQ
 implication        ,X86   ,AVX512DQ_VL          ,AVX512F_VL
 implication        ,X86   ,AVX512VBMI           ,AVX512BW
+implication        ,X86   ,AVX512VBMI_VL        ,AVX512VBMI
 implication        ,X86   ,AVX512VBMI_VL        ,AVX512BW_VL
 
+; While the AVX-512 ISAs can be individually lit-up, they really
```

Comment:
@davidwrighton Could you please review these new implications?

Reply:
These implications don't entirely make sense to me. What they indicate is that if any of avx512cd, bw, f, bmi, or the vl variants are enabled, then all the various instruction sets will be considered to be enabled. If that's the case, then why do we need so many flags? I don't understand. In addition, if you're making the implications have this circular flow that isn't actually required by the flags the processor specifies, we need to treat them similarly in the avx512 detection logic in codeman.cpp. Thus, we can't enable ANY of the AVX512 support in the runtime unless all of the associated cpu flags are enabled.

Reply:
> What they indicate is that if any of avx512cd, bw, f, bmi, or the vl variants are enabled, then all the various instruction sets will be considered to be enabled

Yes and the comment is meant to explain that, if you have any ideas on how to reword it then please feel free to suggest them.

> If that's the case, then why do we need so many flags

We have an actual split in CPUID checks and in other corresponding logic that ties class names to flag names. We may also want to actually allow splitting this in the future, such as if we decide to support Knight's Landing or if some future hardware decides to only support a subset of AVX512.

> In addition, if you're making the implications have this circular flow that isn't actually required by the flags the processor specifies, we need to treat them similarly in the avx512 detection logic in codeman.cpp

This is already handled in codeman by virtue of the:
```
    CPUCompileFlags.Set64BitInstructionSetVariants();
    CPUCompileFlags.EnsureValidInstructionSetSupport();
```

If `Avx512F` was supported but `Avx512BW` was not, then `Avx512F` would be turned off by the `EnsureValidInstructionSetSupport` call.